tty/zynq7000-uart/zynq7000-uart.c

```diff
@@ -267,32 +270,35 @@ static void uart_dispatchMsg(void *arg)
 
 		switch (msg.type) {
 			case mtOpen:
-				break;
-
 			case mtClose:
+				msg.o.err = EOK;
 				break;
 
 			case mtWrite:
-				msg.o.io.err = libtty_write(&uart_common.uart.tty, msg.i.data, msg.i.size, msg.i.io.mode);
+				msg.o.err = libtty_write(&uart_common.uart.tty, msg.i.data, msg.i.size, msg.i.io.mode);
 				break;
 
 			case mtRead:
-				msg.o.io.err = libtty_read(&uart_common.uart.tty, msg.o.data, msg.o.size, msg.i.io.mode);
+				msg.o.err = libtty_read(&uart_common.uart.tty, msg.o.data, msg.o.size, msg.i.io.mode);
 				break;
 
 			case mtGetAttr:
 				if (msg.i.attr.type == atPollStatus) {
 					msg.o.attr.val = libtty_poll_status(&uart_common.uart.tty);
-					msg.o.attr.err = EOK;
+					msg.o.err = EOK;
 					break;
 				}
 
-				msg.o.attr.err = -EINVAL;
+				msg.o.err = -EINVAL;
 				break;
 
 			case mtDevCtl:
 				uart_ioctl(port, &msg);
 				break;
+
+			default:
+				msg.o.err = -EINVAL;
```

Comment:
For consistency sake
```suggestion
				msg.o.err = -ENOSYS;
```